code/modules/vehicles/atv.dm

```diff
@@ -60,3 +63,42 @@
 				turret.pixel_x = 12
 				turret.pixel_y = 4
 				turret.layer = OBJ_LAYER
+
+/obj/vehicle/ridden/atv/attackby(obj/item/W as obj, mob/user as mob, params)
+	if(W.tool_behaviour == TOOL_WELDER && user.a_intent != INTENT_HARM)
+		if(obj_integrity < max_integrity)
+			if(W.use_tool(src, user, 0, volume=50, amount=1))
+				user.visible_message("<span class='notice'>[user] repairs some damage to [name].</span>", "<span class='notice'>You repair some damage to [src].</span>")
+				obj_integrity += min(10, max_integrity-obj_integrity)
+				if(obj_integrity == max_integrity)
+					to_chat(user, "<span class='notice'>It looks to be fully repaired now.</span>")
+		return TRUE
+	return ..()
+
+/obj/vehicle/ridden/secway/obj_break()
+	START_PROCESSING(SSobj, src)
+	return ..()
+
+/obj/vehicle/ridden/atv/process()
+	if(obj_integrity >= integrity_failure || obj_integrity <= 0)
```

Comment:
if it's below 0 it should be deleting (which kills process already) shouldn't it?